### ISSUE TITLE
Should the order in which the compiler reports diagnostics be deterministic?

### ISSUE THREAD

**Issue by amcasey** [dotnet/roslyn#1506]
I'm most interested in the command-line scenario but I guess you could ask the same question about the API.

**Comment by amcasey** [comment]
FYI @gafter

**Comment by jaredpar** [comment]
Roslyn needs to consider itself as both a command line compiler and a compiler service.  If determinism is important from the command line it seems logical it should be important from the service as well.  Otherwise it's a very limited feature.

**Comment by theoy** [comment]
I agree with @jaredpar - ultimately to declare success on Determinism, all observable effects as part of the contract should be deterministic. We have a new error log format for example, and the output should be bitwise identical if the compiler was provided identical inputs.

**Comment by jaredpar** [comment]
Discussed this a while back with @gafter and @tmat and the conclusion was yes this order should be deterministic.

**Comment by mavasani** [comment]
A side effect of this non-determinism is that the SARIF error log file generated with /errorlog switch (containing all diagnostics in JSON format) may have diagnostics logged in different order for the same command line across builds.

@michaelcfanning @lgolding @morrisjoe is this important enough scenario to fix for 1.1 milestone?

**Comment by lgolding** [comment]
From the point of view of SARIF itself, the order in which issues appear within the log file is not significant. That is, there is nothing in the SARIF spec that specifies the order, or allows consumers to draw any conclusions from the order.
